### DISCUSSION TITLE
how to get exif information

### DISCUSSION THREAD

**Discussion by powerpistn** [ImageMagick/ImageMagick#5168] [Consulting]
I want to get the picture information about EXIF。 I build the GraphicsMagick。my code is as follows:
`
#include<iostream>
#include<map>

using namespace std;

#include <Magick++.h>
using  namespace Magick;
int main()
{
    InitializeMagick((const char *)nullptr);
    Image img;
   img.read("/untitled2/111.jpeg");
    cout<<"11111"<<endl;
    std::string value=img.attribute("EXIF:Depth").c_str();
    cout<<"value="<<value<<endl;
    cout<<"222"<<endl;
    return  0;
}

`
however， I got nothing. The results of the program
`
11111
value=
222
`

when I use "gm identify -verbose 111.jpeg", the result is as follows
`
Image: 111.jpeg
  Format: JPEG (Joint Photographic Experts Group JFIF format)
  Geometry: 194x259
  Class: DirectClass
  Type: true color
  Depth: 8 bits-per-pixel component
  Channel Depths:
    Red:      8 bits
    Green:    8 bits
    Blue:     8 bits
  Channel Statistics:
    Red:
      Minimum:                     0.00 (0.0000)
      Maximum:                   255.00 (1.0000)
      Mean:                       81.37 (0.3191)
      Standard Deviation:         71.20 (0.2792)
    Green:
      Minimum:                     0.00 (0.0000)
      Maximum:                   255.00 (1.0000)
      Mean:                       92.22 (0.3617)
      Standard Deviation:         63.38 (0.2485)
    Blue:
      Minimum:                     0.00 (0.0000)
      Maximum:                   255.00 (1.0000)
      Mean:                      105.17 (0.4124)
      Standard Deviation:         66.33 (0.2601)
  Filesize: 6.2Ki
  Interlace: No
  Orientation: Unknown
  Background Color: white
  Border Color: #DFDFDF
  Matte Color: #BDBDBD
  Page geometry: 194x259+0+0
  Compose: Over
  Dispose: Undefined
  Iterations: 0
  Compression: JPEG
  JPEG-Quality: 71
  JPEG-Colorspace: 2
  JPEG-Colorspace-Name: RGB
  JPEG-Sampling-factors: 2x2,1x1,1x1
  Signature: 0180f3270be4b8b94480b7234e7e5895e88b78e25d4cd20dce25957e660d4899
  Tainted: False
  Elapsed Time: 0m:0.002059s
  Pixels Per Second: 23.3Mi

`
I want to know how to get the EXIF information using the C++ program?

**Comment by snibgo** [comment]
This forum is for ImageMagick ("IM"), not GraphicsMagick ("GM").

Your "identify verbose" shows that your file has no "EXIF:Depth" property. So your C++ program correctly shows no value.
